scriptworker/github.py

```diff
@@ -0,0 +1,90 @@
+"""GitHub helper functions."""
```

Comment:
I have a couple of suggestions about the code here:
- I think the functions here should be at a higher level of abstraction. In other words, they should fetch the data themselves, and provide it in a useful form. That way, `.cot.verify` doesn't need to know details of the structure of github's API, or the results it generates.
- I would consider using an existing binding to the github api, such as [github3.py](https://github3py.readthedocs.io/en/stable/), rather than rolling our own. (Though still possibly worth while to wrap it here).

Reply:
Having looked at the changes in `.cot.verify`, it looks like these are use mainly to synthesize github generated events. One thought that occurred to me (though I'm not convinced it is correct) is that functions here should do the work of synthesizing the event data, and the code in `.cot.verify` can wrap that to create the context.

If we do go this direction, I suspect that the cron context should be handled differently, since the pseudo-event isn't actually related to github.

Reply:
Done. I used github3.py, even that's not an async library. The reason is: none of [the highlighted libraries](https://developer.github.com/v3/libraries/) is async. If we want to make this module async, we could use [aioify](https://github.com/yifeikong/aioify). However, the lib is quite unstable and I'm not certain it's worth the effort yet => we don't parallelize any call to github in `cot.verify`. 

Regarding the cron context, we can define something that doesn't try to be a github event. It was implemented this way because it allows some factorization in https://github.com/mozilla-mobile/android-components/blob/b2c827e66d89f25688a15e7b27f7494ad2c718dc/.taskcluster.yml#L153 and in the scripts that handles the decision task. If we want to revisit this context, I would suggest to do it in a follow up.